api/v1alpha1/openapi.yaml

```diff
@@ -1577,21 +1577,26 @@ components:
       properties:
```

Comment:
(Comment does not belong here, I don't know where it belongs)

When trying to create a new source of type `InlineConfigProviderSpec`, it is created, but we receive an invalid condition but no much info about why. Is there some issue on the request, or any idea why this is happening?

Request:
```
{
  "apiVersion": "v1alpha1",
  "kind": "Fleet",
  "metadata": {
    "name": "name",
    "labels": {}
  },
  "spec": {
    "selector": {
      "matchLabels": {}
    },
    "template": {
      "metadata": {
        "labels": {
          "fleet": "name"
        }
      },
      "spec": {
        "os": {
          "image": "some-image"
        },
        "config": [
          {
            "configType": "InlineConfigProviderSpec",
            "inline": {
              "one": "a",
              "two": "b"
            },
            "name": "my-config"
          }
        ]
      }
    }
  }
}
```

Response:
```
  "status": {
    "conditions": [
      {
        "lastTransitionTime": "2024-06-14T15:06:34.509800963Z",
        "message": "fleet has 1 invalid configuration: my-config",
        "reason": "Invalid",
        "status": "False",
        "type": "Valid"
      }
    ]
  }
```

Reply:
The reason is `config[0].inline` must be valid Ignition (see examples/fleet.yaml). I think the message could be a bit more helpful in that respect.

Reply:
We'll add more validations in the UI for this particular case, but I think @avishayt 's proposal would help better understand other errors.

Reply:
I opened [MGMT-18179](https://issues.redhat.com/browse/MGMT-18179) to track it.  It's out of scope for this PR.